soiree/app/controllers/group/playlists_controller.rb

```diff
@@ -16,17 +16,21 @@ def create
 
   def destroy
     playlist = Playlist.find(params[:id])
-
-    delete_group_redirect(playlist)
+    delete_group_invites(playlist)
+    redirect_to dashboard_path
   end
 
   private
 
-  def delete_group_redirect(playlist)
-    playlist.group_playlists.destroy_all
-    playlist.playlist_songs.destroy_all
-    playlist.destroy
-    redirect_to dashboard_path
+  def delete_group_invites(playlist)
+    group = playlist.groups[0]
```

Comment:
I think having nil in a field is maybe not the worst thing? Then you can have a `belongs to` relationship, and use `if playlist.group` if you need to confirm that it has one, because it would return a boolean. But having not seen the rest of the code base, this might have broader implications than I'm imagining. Your call